tests/capture/test_operators.py

```diff
@@ -0,0 +1,301 @@
+        else:
+            qml.X(w)
+
+    jaxpr = jax.make_jaxpr(qfunc)()
+
+    assert len(jaxpr.eqns) == 1
+
+    eqn = jaxpr.eqns[0]
+    assert eqn.primitive == qml.X._primitive
+    assert len(eqn.invars) == 1
+    assert isinstance(eqn.invars[0], jax.core.Literal)
+    assert eqn.invars[0].val == 0
+
+    assert isinstance(eqn.outvars[0].aval, AbstractOperator)
+    assert isinstance(eqn.outvars[0], jax.core.DropVar)
+
+    assert eqn.params == {"n_wires": 1}
+
+    qml.capture.disable_plxpr()
+
+
+def test_parametrized_op():
+    """Test capturing a parametrized operation."""
+    qml.capture.enable_plxpr()
+
+    jaxpr = jax.make_jaxpr(qml.Rot)(1.0, 2.0, 3.0, 10)
+
+    assert len(jaxpr.eqns) == 1
+    eqn = jaxpr.eqns[0]
+
+    assert eqn.primitive == qml.Rot._primitive
+    assert len(eqn.invars) == 4
+    assert jaxpr.jaxpr.invars == jaxpr.eqns[0].invars
+
+    assert isinstance(eqn.outvars[0].aval, AbstractOperator)
+    assert eqn.params == {"n_wires": 1}
+
+    qml.capture.disable_plxpr()
+
+
+def test_pauli_rot():
+    """Test a special operation that has positional metadata and overrides binding."""
+
+    qml.capture.enable_plxpr()
+
+    def qfunc(a, wire0, wire1):
+        qml.PauliRot(a, "XY", (wire0, wire1))
+
+    jaxpr = jax.make_jaxpr(qfunc)(0.5, 2, 3)
+    assert len(jaxpr.eqns) == 1
+    eqn = jaxpr.eqns[0]
+
+    assert eqn.primitive == qml.PauliRot._primitive
+    assert eqn.params == {"pauli_word": "XY", "id": None, "n_wires": 2}
+
+    assert len(eqn.invars) == 3
+    assert jaxpr.jaxpr.invars == eqn.invars
+
+    qml.capture.disable_plxpr()
+
+
+class TestTemplates:
+    def test_variable_wire_non_parametrized_template(self):
+        """Test capturing a variable wire, non-parametrized template like GroverOperator."""
+
+        qml.capture.enable_plxpr()
+
+        jaxpr = jax.make_jaxpr(qml.GroverOperator)(wires=(0, 1, 2, 3, 4, 5))
+
+        assert len(jaxpr.eqns) == 1
+        eqn = jaxpr.eqns[0]
+
+        assert eqn.primitive == qml.GroverOperator._primitive
+        assert eqn.params == {"n_wires": 6}
+        assert eqn.invars == jaxpr.jaxpr.invars
+
+        qml.capture.disable_plxpr()
+
+    def test_nested_template(self):
+        """Test capturing a template that contains a nested opeartion defined outside the qfunc."""
+
+        qml.capture.enable_plxpr()
+
+        coeffs = [0.25, 0.75]
+        ops = [qml.X(0), qml.Z(0)]
+        H = qml.dot(coeffs, ops)
+
+        def qfunc(Hi):
+            qml.TrotterProduct(Hi, time=2.4, order=2)
+
+        jaxpr = jax.make_jaxpr(qfunc)(H)
+
+        assert len(jaxpr.eqns) == 6
+
+        # due to flattening and unflattening H
+        assert jaxpr.eqns[0].primitive == qml.X._primitive
+        assert jaxpr.eqns[1].primitive == qml.ops.SProd._primitive
+        assert jaxpr.eqns[2].primitive == qml.Z._primitive
+        assert jaxpr.eqns[3].primitive == qml.ops.SProd._primitive
+        assert jaxpr.eqns[4].primitive == qml.ops.Sum._primitive
```

Comment:
Could add a jaxpr-level check that no ops are "queued"
```suggestion
        assert jaxpr.eqns[4].primitive == qml.ops.Sum._primitive
        assert not any(isinstance(eqn.outvars[0], jax.core.DropVar) for eqn in jaxpr.eqns[:5])
```